internal/rbd/encryption.go

```diff
@@ -74,6 +77,30 @@ const (
 	luksSlot1 = "1"
 )
 
+// getLuksHeaderSizeMetadata retrieves the LUKS header size from the
+// image metadata. Older images (<=3.14) did not set the metadata
+// luks2HeaderSizeKey, so the default value is returned in that case
+// i.e, DefaultLuks2HeaderSize.
+//
+//nolint:unused // Unused code will be used in future.
+func (ri *rbdImage) getLuksHeaderSizeMetadata() (uint64, error) {
```

Comment:
It would be good to return an error in case the `rbdImage` is not encrypted, or maybe no error and size=0? It will prevent incorrect usage and taking a non-existing offset into account.